azure-pipelines.yml

```diff
@@ -0,0 +1,761 @@
+  steps:
+  - task: DownloadBuildArtifacts@0
+    displayName: 'Download Metadata'
+    inputs:
+      artifactName: metadata
+
+
+  - task: DownloadBuildArtifacts@0
+    displayName: 'Download PyPI Packages'
+    inputs:
+      artifactName: pypi
+
+
+  - bash: |
+       #!/usr/bin/env bash
+
+       # Verify the pip wheels
+
+       set -ex
+
+       CLI_VERSION=`cat $SYSTEM_ARTIFACTSDIRECTORY/metadata/version`
+       PYPI_FILES=$(cd $SYSTEM_ARTIFACTSDIRECTORY/pypi; pwd)
+
+
+       echo "== Testing pip install on Python 3.6 =="
+       docker run \
+         --rm -v $PYPI_FILES:/mnt/pypi python:3.6 \
+         /bin/bash -c "ls /mnt/pypi && pip install -f /mnt/pypi -q azure-cli==$CLI_VERSION && az self-test && az --version && sleep 5"
+
+       echo "== Testing pip install on Python 3.7 =="
+       docker run \
+         --rm -v $PYPI_FILES:/mnt/pypi python:3.7 \
+         /bin/bash -c "ls /mnt/pypi && pip install -f /mnt/pypi -q azure-cli==$CLI_VERSION && az self-test && az --version && sleep 5"
+
+       echo "== Testing pip install on Python 2.7 =="
+       docker run \
+         --rm -v $PYPI_FILES:/mnt/pypi python:2.7 \
+         /bin/bash -c "ls /mnt/pypi && pip install -f /mnt/pypi -q azure-cli==$CLI_VERSION && az self-test && az --version && sleep 5"
+
+    displayName: 'Test pip Install'
+
+
+- phase: Phase_14
+  displayName: Build Homebrew Formula
+
+  dependsOn: Phase_1
+  condition: succeeded()
+  queue:
+    name: Hosted Ubuntu 1604
+#Your build pipeline references an undefined variable named ‘az storage blob url -c $CONTAINER_NAME -n $TAR_NAME -otsv’. Create or edit the build pipeline for this YAML file, define the variable on the Variables tab. See https://go.microsoft.com/fwlink/?linkid=865972
+#Your build pipeline references an undefined variable named ‘cd $(dirname $0’. Create or edit the build pipeline for this YAML file, define the variable on the Variables tab. See https://go.microsoft.com/fwlink/?linkid=865972
+#Your build pipeline references an undefined variable named ‘cd $BUILD_SOURCESDIRECTORY/scripts/release/homebrew/docker; pwd’. Create or edit the build pipeline for this YAML file, define the variable on the Variables tab. See https://go.microsoft.com/fwlink/?linkid=865972
+#Your build pipeline references an undefined variable named ‘cd $BUILD_SOURCESDIRECTORY/scripts/release/homebrew/docker; pwd’. Create or edit the build pipeline for this YAML file, define the variable on the Variables tab. See https://go.microsoft.com/fwlink/?linkid=865972
+#Your build pipeline references an undefined variable named ‘cd $SYSTEM_ARTIFACTSDIRECTORY/pypi; pwd’. Create or edit the build pipeline for this YAML file, define the variable on the Variables tab. See https://go.microsoft.com/fwlink/?linkid=865972
```

Comment:
What's up with these comments?

Reply:
A little bit unclear, seeing as they would suggest that the original definition shouldn't be working, or at very least is stashing intermediate progress in the wrong places. However, I've updated them to refer to variables that are defined by DevOps in d7ba697.

Reply:
Should the previous commit have failed the build then?  They were successful despite this warning.

Reply:
I don't _believe_ they should have failed. I suspect in each case, the empty string was being substituted for the expected value, and stashing the intermediate build products somewhere else proved unproblematic. Updating these values will drive consistency though.